### PR TITLE
Fix a couple of issues

### PR DESCRIPTION
**Pull request by armenzg**
This is simply to reduce alerting.

### CODE REVIEW

File: gitbot/deployhook.py [hunk 1]
```diff
@@ -193,6 +193,8 @@ def process_pull_request() -> tuple[str, int]:
 
         if pull_request["merged"]:
             return respond("Pull request is already merged.", status_code=200)
+    elif ENV == "staging":
+        return respond("We do not support this for staging.", status_code=200)
```
Comment (armenzg) [inline gitbot/deployhook.py:197]: A developer opens a PR on each repo but the branch does not exist on `getsentry-test-repo`.
This is why we get alert like this:
image

File: gitbot/lib.py [hunk 2]
```diff
@@ -169,8 +169,16 @@ def bump_version(
...
 
-        command = bump_command(ref_sha, author)
-        run(command, cwd=repo_root)
+        try:
+            command = bump_command(ref_sha, author)
+            run(command, cwd=repo_root)
+        except CommandError:
+            execution = run("git show", cwd=repo_root)
+            # e.g. https://github.com/getsentry/getsentry/pull/7672/commits
+            if execution.stdout.find(f"getsentry/sentry@{ref_sha}") > -1:
```
Comment (armenzg) [inline gitbot/lib.py:178]: This doesn't happen often but I would rather not get anyone distracted:
https://sentry.io/organizations/sentry/issues/2677689153/?environment=production&project=5748916&query=&statsPeriod=90d

File: gitbot/lib.py [hunk 3]
```diff
@@ -169,8 +169,16 @@ def bump_version(
...
+        try:
+            command = bump_command(ref_sha, author)
+            run(command, cwd=repo_root)
+        except CommandError as e:
+            execution = run("git show", cwd=repo_root)
+            # e.g. https://github.com/getsentry/getsentry/pull/7672/commits
+            if execution.stdout.find(f"getsentry/sentry@{ref_sha}") > -1:
+                logger.info("The developer has manually bumped the version.")
+            else:
+                raise e
```
Comment (asottile-sentry) [inline gitbot/lib.py:181]: reraise in python should just be `raise` with no arguments: https://youtu.be/yj4z1N3_0sw

File: gitbot/lib.py [hunk 4]
```diff
@@ -169,8 +169,16 @@ def bump_version(
...
 
-        command = bump_command(ref_sha, author)
-        run(command, cwd=repo_root)
+        try:
+            command = bump_command(ref_sha, author)
+            run(command, cwd=repo_root)
+        except CommandError as e:
+            execution = run("git show", cwd=repo_root)
+            # e.g. https://github.com/getsentry/getsentry/pull/7672/commits
+            if execution.stdout.find(f"getsentry/sentry@{ref_sha}") > -1:
```
Comment (asottile-sentry) [inline gitbot/lib.py:178]: seems fine -- alternatively we could also make the command in `getsentry` [idempotent](https://youtu.be/o9THkT5ZPi4) by first checking `git diff --quiet` before attempting to commit nothing

**Review by asottile-sentry — APPROVED** [review]

File: gitbot/lib.py [hunk 5]
```diff
@@ -172,13 +172,13 @@ def bump_version(
         try:
             command = bump_command(ref_sha, author)
             run(command, cwd=repo_root)
-        except CommandError as e:
+        except CommandError as e:  # noqa: F841
```
Comment (asottile-sentry) [inline gitbot/lib.py]: should be able to remove the `as e` -- that's what `flake8` is telling you here

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**